### DISCUSSION TITLE
when I package a theia plugin use "vsce package", have error "Manifest missing field: engines.vscode", how can I package the plugin to vsix

### DISCUSSION THREAD

**Discussion by ReachForAStar** [eclipse-theia/theia#8561] [General] [answered]
It's the helloworld plugin,package.json is this

```json
{
  "name": "helloword",
  "publisher": "theia",
  "keywords": [
    "theia-plugin"
  ],
  "version": "0.0.1",
  "license": "none",
  "files": [
    "src"
  ],
  "activationEvents": [
    "*"
  ],
  "devDependencies": {
    "@theia/plugin": "next",
    "@theia/plugin-packager": "0.0.1-1594417672",
    "rimraf": "2.6.2",
    "typescript": "3.5.3",
    "typescript-formatter": "7.2.2"
  },
  "scripts": {
    "prepare": "yarn run clean && yarn run build",
    "clean": "rimraf lib",
    "format-code": "tsfmt -r",
    "watch": "tsc -watch",
    "compile": "tsc",
    "build": "yarn run format-code && yarn run compile && theia-plugin pack"
  },
  "engines": {
    "theiaPlugin": "next"
  },
  "theiaPlugin": {
    "backend": "lib/helloword-backend.js"
  }
}
```

**Comment by vince-fugnitto** [comment, marked as answer]
@ReachForAStar thank you for the discussion, the error message clearly states that an `engines.vscode` field is missing from your `package.json`, for example:

```json
"engines": {
  "vscode": "^1.40.0"
},
```

**Reply by ReachForAStar** [reply]
Thank  you for the answer, it's useful for me.  I'm worry engines changed maybe have some influence on the theia plugin.

**Reply by vince-fugnitto** [reply]
> Thank you for the answer, it's useful for me. I'm worry engines changed maybe have some influence on the theia plugin.

Just for additional information, the framework defines a default supported VS Code API version (which can be modified).
If you want compatibility, you should choose a version which is support (hence why I referenced `^1.40.0`).

https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts#L22
